cmd/oras/root/push.go

```diff
@@ -130,12 +136,19 @@ Example - Push file "hi.txt" into an OCI image layout folder 'layout-dir' with t
 					}
 				}
 			}
+			configAndPlatform := []string{"config", "artifact-platform"}
+			if err := oerrors.CheckMutuallyExclusiveFlags(cmd.Flags(), configAndPlatform...); err != nil {
+				return err
+			}
 
 			switch opts.PackVersion {
 			case oras.PackManifestVersion1_0:
 				if opts.manifestConfigRef != "" && opts.artifactType != "" {
 					return errors.New("--artifact-type and --config cannot both be provided for 1.0 OCI image")
 				}
+				if opts.Platform.Platform != nil && opts.artifactType != "" {
+					return errors.New("--artifact-platform and --artifact-type cannot both be provided for 1.0 OCI image")
+				}
```

Comment:
I'm not sure what the v1.0 rule should be.

Reply:
`--artifact-platform` specifies content of config blob, `--artifact-type` specifies `mediaType` of config blob, so both flags can be used for OCI v1.0 image.

Reply:
```suggestion
```

Reply:
Since the code above this does not allow specifying `--artifact-type` and `--config` at the same time I thought it made sense to not allow `--artifact-typ` and `--artifact-platform` since platform is just a shortcut for config.

Reply:
We do not allow specifying `--artifact-type` and `--config` at the same time because **those flags function are overlapped, user might specify different media types of config descriptor**, like `--artifact-type aaa --config xxx:bbb`, `aaa` and `bbb` conflicts.

But there is no such potential conflict between `--artifact-type` and `--artifact-platform`